logging/src/lib.rs

```diff
@@ -133,7 +132,7 @@ impl<T> Logger<T> {
     /// This implementation borrows a shared (but thread-local) buffer of log events, to ensure
     /// that the `action` only sees one stream of events with increasing timestamps. This may
     /// have a cost that we don't entirely understand.
-    pub fn log<S: Into<T>>(&self, event: S) {
+    pub fn log<S>(&self, event: S) where CB: PushInto<(Duration, S)> {
```

Comment:
Can you say why the `Into<T>` went away? It feels like a bit of the noise in other files is making up for this. Is it that we shouldn't want to invoke `.into()` before pushing, or .. some other reason? Or no particular reason?